server/find/server.go

```diff
@@ -516,3 +516,15 @@ func createExtendedProviderResult(epInfo registry.ExtendedProviderInfo, iVal ind
 		},
 	}
 }
+
+func caseMHString(s string) (multihash.Multihash, error) {
+	mh, err := multihash.FromHexString(s)
```

Comment:
My sense is that this diverging from the IPFS standards of always using base-58 encoded string value of mutlihash in URLs (for good reason). 

If the demand for this feature is popular enough, ipni spec update is the place to start. Otherwise my vote would be to keep it consistent and allow lookup by base-58 encoding only, similar to pretty much every IPFS api out there?

Reply:
Ideally, I agree with keeping it as base58. However there are many tools that show multihashes as hex-encoded strings, and allowing these into the API will increase interoperability, even if just pasting multihashes in to curl.

Added logging to log both decoding errors, and unit test.

Reply:
OK 👍 

Could we make sure spec is also updated please?